### ISSUE TITLE
dump/load gives Invalid input type

### ISSUE THREAD

**Issue by neggert** [lidatong/dataclasses-json#65]
Using `schema().load()` does not seem to work for classes that contain lists. I can reproduce this even with the Boss/Minion example from the documentation

```{python}
@dataclass_json
@dataclass(frozen=True)
class Minion:
    name: str


@dataclass_json
@dataclass(frozen=True)
class Boss:
    minions: List[Minion]

boss = Boss([Minion('evil minion'), Minion('very evil minion')])
boss_dict = Boss.schema().dump(boss)
Boss.schema().load(boss_dict)
```

```
marshmallow.exceptions.ValidationError: {'_schema': ['Invalid input type.']}
```

For what it's worth, going all the way to JSON and back seems to work fine:

```
boss_json = boss.to_json()
Boss.from_json(boss_json)
```

**Comment by nuklea** [comment]
Same issue with 0.2.1

**Comment by nuklea** [comment]
```python
from dataclasses import dataclass
from typing import List

from dataclasses_json import DataClassJsonMixin


@dataclass
class Entity(DataClassJsonMixin):
    id: int


@dataclass
class Container(DataClassJsonMixin):
    entities: List[Entity]


@dataclass
class Data(DataClassJsonMixin):
    container: Container


schema = Data.schema()
schema.load({'container': {'entities': [{'id': 1}]}})
```

raises `marshmallow.exceptions.ValidationError: {'container': {'entities': {'_schema': ['Invalid input type.']}}}`

**Comment by nuklea** [comment]
But works fine with 0.1.0

**Comment by lidatong** [comment]
Thanks for reporting this. Fixed in #66
